src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1365,6 +1365,7 @@ private string ReadLineFromFile(string initialContent)
                 }
 
                 var c = unchecked((char)inC);
+                if (c == '\b' && !NoPrompt && sb.Length == 0) continue;
```

Comment:
I'd expect the check in line below `if (c == '\b' && !NoPrompt && && sb.Length > 0)

Reply:
What do you mean?
After the Console.Out.Write?
The user of this code have only the prompt at this stage, so echoing him the backspace will make him think the prompt was deleted.

Reply:
It seems that this is really an edge case where there is nothing in the stringbuilder so a backspace in this case is the same as being at the prompt, typing backspace and nothing happens.  In general, I would agree we should not filter the input stream, but this specific case seems ok.

Reply:
@bugale Please follow our code convention:
```suggestion
                if (c == '\b' && !NoPrompt && sb.Length == 0)
                {
                    continue;
                }
```

Reply:
Okay, but exactly one line below that there is a code that uses this conventions

Reply:
Skipping braces is dangerous. We should use only good patterns in new code. We don't request to fix all old code in the method but you can do this if you want.

Reply:
Yeah, I know Apple's "goto fail" :) Ill change the other one too